neo/rawio/axonarawio.py

```diff
@@ -81,57 +98,149 @@ def _parse_header(self):
-        self.data_dtype = np.int16
-
-        # There is no global header for .bin files
-        self.global_header_size = 0
-
-        # Generally useful information
-        self.sr = int(params['rawRate'])
-        self.num_channels = len(self.get_active_tetrode()) * 4
-
-        # How many 432byte packets does this data contain (<=> num. samples/3)?
-        self.num_total_packets = int(os.path.getsize(self.bin_file)
-                                     / self.bytes_packet)
-        self.num_total_samples = self.num_total_packets * 3
-
-        # Create np.memmap to .bin file
-        self._raw_signals = np.memmap(
-            self.bin_file, dtype=self.data_dtype, mode='r',
-            offset=self.global_header_size
-        )
-
-        # Header dict
+        unit_dtype = np.dtype([('spiketimes', '>i4'),
+                               ('samples', 'int8', (50,))])
+
+        # Utility collection of file parameters (general info and header data)
+        params = {'bin': {'filename': self.bin_file,
+                          'bytes_packet': 432,
+                          'bytes_data': 384,
+                          'bytes_head': 32,
+                          'bytes_tail': 16,
+                          'data_type': 'int16',
+                          'header_size': 0,
+                          # bin files don't contain a file header
+                          'header_encoding': None},
+                  'set': {'filename': self.set_file,
+                          'header_encoding': 'cp1252'},
+                  'unit': {'data_type': unit_dtype,
+                           'tetrode_ids': [],
+                           'header_encoding': 'cp1252'}}
+        self.file_parameters = params
+
+        # SCAN SET FILE
+        set_dict = self.get_header_parameters(self.set_file, 'set')
+        params['set']['file_header'] = set_dict
+        params['set']['sampling_rate'] = int(set_dict['rawRate'])
+
+        # SCAN BIN FILE
+        signal_streams = []
+        signal_channels = []
+        if self.bin_file:
+            bin_dict = self.file_parameters['bin']
+            # add derived parameters from bin file
+            bin_dict['num_channels'] = len(self.get_active_tetrode()) * 4
+            num_tot_packets = int(self.bin_file.stat().st_size
+                                  / bin_dict['bytes_packet'])
+            bin_dict['num_total_packets'] = num_tot_packets
+            bin_dict['num_total_samples'] = num_tot_packets * 3
+
+            # Create np.memmap to .bin file
+            self._raw_signals = np.memmap(
+                self.bin_file, dtype=self.file_parameters['bin']['data_type'],
+                mode='r', offset=self.file_parameters['bin']['header_size']
+            )
+
+            signal_streams = self._get_signal_streams_header()
+            signal_channels = self._get_signal_chan_header()
+
+        # SCAN TETRODE FILES
+        # In this IO one tetrode corresponds to one unit
```

Comment:
I don't anderstand this.
Do you mean that spike here are unsorted so all spikes per tetrode belong to the same units ?

Reply:
When reading data in `.X` format (tetrode files) that were generated from the raw recording (recommended default case according to the Axona manual), then yes, the units are unsorted. Or rather, the only thing that happened is that a threshold was applied to the raw data to find 'spikes'. Then only data surrounding those spikes was written to files and all data discarded. Each spike timestamp is the same for all channels on a given tetrode. There is also a mechanism for validating these spikes - they should not be too close together, but I am not sure of the details there from the top of my head. Since this is something that the Axona software does as a standard processing step it makes to assume this here.

However, we are now also at the point where we can export spike sorted data (from spikeinterface) to the same `.X` format, so it makes sense to check if that can still be read in with this rawio.

Reply:
I extended the comment to add the note about the tetrode not being sorted yet at that stage.